src/app/pages/journal/journal.page.ts

```diff
@@ -204,36 +210,33 @@ export class JournalPage extends BasePageComponent implements OnInit {
     this.completedTests = completedTests;
   };
 
-  // handleLoadingUI = (isLoading: boolean): void => {
-  //   if (isLoading) {
-  //     this.loadingSpinner = this.loadingController.create({
-  //       dismissOnPageChange: true,
-  //       spinner: 'circles',
-  //     });
-  //     this.loadingSpinner.present();
-  //     return;
-  //   }
-  //   this.pageRefresher ? this.pageRefresher.complete() : null;
-  //   if (this.loadingSpinner) {
-  //     this.loadingSpinner.dismiss();
-  //     this.loadingSpinner = null;
-  //   }
-  // }
+  handleLoadingUI = async (isLoading: boolean, loadingSpinner$: Observable<HTMLIonLoadingElement>): Promise<void> => {
+    const spinner = await loadingSpinner$.toPromise();
+    if (isLoading) {
+      await spinner.present();
+      return;
+    }
+    if (this.pageRefresher) {
+      await this.pageRefresher.complete();
+    }
+    if (spinner) {
+      await spinner.dismiss();
+    }
+  };
 
   showError = (error: MesError): void => {
     if (error === undefined || error.message === '') return;
-
     // Modals are at the same level as the ion-nav so are not getting the zoom level class,
     // this needs to be passed in the create options.
 
-    // TODO figure out what to do about zoom methods not existing in appComponent
-    // const zoomClass = `modal-fullscreen ${this.app.getTextZoomClass()}`;
```

Comment:
`this.app.getTextZoomClass()` should now have been implemented so we should be able to use that here